src/main/java/de/tum/in/www1/exerciseapp/domain/MultipleChoiceQuestionStatistic.java

```diff
@@ -0,0 +1,214 @@
+                    if(mcSubmittedAnswer.getSelectedOptions().contains(answerCounter.getAnswer())) {
+                        answerCounter.setUnRatedCounter(answerCounter.getUnRatedCounter()+1);
+                    }
+                }
+            }
+
+            // increase correctCounter if answer is complete correct
+            if(getQuestion().isAnswerCorrect(mcSubmittedAnswer)) {
+                setUnRatedCorrectCounter(getUnRatedCorrectCounter()+1);
+            }
+        }
+    }
+
+    /**
+     * 1. check if the Result is rated or unrated
+     * 2. decrease participants, all selected AnswerCounter and if the question is correct, than decrease the correctCounter
+     *
+     * @param submittedAnswer the submittedAnswer object which contains all selected answers
+     * @param rated specify if the Result was rated ( participated during the releaseDate and the dueDate of the quizExercise)
+     *                                  or unrated  ( participated after the dueDate of the quizExercise)
+     */
+    @Override
+    public void removeOldResult(SubmittedAnswer submittedAnswer, boolean rated) {
+
+        if(submittedAnswer == null) {
+            return;
+        }
+
+        MultipleChoiceSubmittedAnswer mcSubmittedAnswer = (MultipleChoiceSubmittedAnswer)submittedAnswer;
+
+        if(rated) {
+            //decrease participants
+            setParticipantsRated(getParticipantsRated()-1);
+
+            if(mcSubmittedAnswer.getSelectedOptions() != null) {
+                // decrease answerCounter if answer is selected
+                for (AnswerCounter answerCounter: answerCounters) {
+                    if(mcSubmittedAnswer.getSelectedOptions().contains(answerCounter.getAnswer())) {
+                        answerCounter.setRatedCounter(answerCounter.getRatedCounter()-1);
+                    }
+                }
+            }
+            // decrease correctCounter if answer is complete correct
+            if(getQuestion().isAnswerCorrect(mcSubmittedAnswer)) {
+                setRatedCorrectCounter(getRatedCorrectCounter()-1);
+            }
+        }
+        // Result is unrated
+        else{
+            //decrease participants
+            setParticipantsUnrated(getParticipantsUnrated()-1);
+
+            if(mcSubmittedAnswer.getSelectedOptions() != null) {
+                // decrease answerCounter if answer is selected
+                for (AnswerCounter answerCounter: answerCounters) {
+                    if(mcSubmittedAnswer.getSelectedOptions().contains(answerCounter.getAnswer())) {
+                        answerCounter.setUnRatedCounter(answerCounter.getUnRatedCounter()-1);
+                    }
+                }
+            }
+            // decrease correctCounter if answer is complete correct
+            if(getQuestion().isAnswerCorrect(mcSubmittedAnswer)) {
+                setUnRatedCorrectCounter(getUnRatedCorrectCounter()-1);
+            }
+        }
+    }
```

Comment:
again it seems the code for rated and unrated is the same. Can you try to extract a method out of it to avoid code duplication?